peregrine/iqgen/iqgen_main.py

```diff
@@ -0,0 +1,460 @@
+          namespace.amplitude_value1 = 0.5
+        if namespace.amplitude_period is None:
+          namespace.amplitude_period = 0.5
+
+        amplitude = AmplitudeSine(namespace.amplitude_value0,
+                                  namespace.amplitude_value1,
+                                  namespace.amplitude_period)
+      else:
+        raise ValueError("Unsupported amplitude type")
+      sv.setAmplitude(amplitude)
+
+  class UpdateMessageType(UpdateSv):
+    def __init__(self, option_strings, dest, nargs=None, **kwargs):
+      super(UpdateMessageType, self).__init__(option_strings, dest, **kwargs)
+
+    def doUpdate(self, sv, parser, namespace, values, option_string):
+      if namespace.message_type == "zero":
+        message = ConstMessage(1)
+      elif namespace.message_type == "one":
+        message = ConstMessage(-1)
+      elif namespace.message_type == "zero+one":
+        message = ZeroOneMessage()
+      else:
+        raise ValueError("Unsupported message type")
+      sv.setL1CAMessage(message)
+      sv.setL2CMessage(message)
+
+  class UpdateMessageFile(UpdateSv):
+    def __init__(self, option_strings, dest, nargs=None, **kwargs):
+      super(UpdateMessageFile, self).__init__(option_strings, dest, **kwargs)
+
+    def doUpdate(self, sv, parser, namespace, values, option_string):
+      data = numpy.fromfile(namespace.message_file, dtype=numpy.uint8)
+      namespace.message_file.close()
+      data = numpy.unpackbits(data)
+      data = numpy.asarray(data, dtype=numpy.int8)
+      data <<= 1
+      data -= 1
+      numpy.negative(data, out=data)
+      message = BlockMessage(data)
+
+      sv.setL1CAMessage(message)
+      sv.setL2CMessage(message)
+
+  parser = argparse.ArgumentParser(description="Signal generator", usage='%(prog)s [options]')
+  parser.add_argument('--gps-sv',
+                      default=[],
+                      help='Enable GPS satellite',
+                      action=AddSv)
+  parser.add_argument('--bands',
+                      default="l1ca",
+                      choices=["l1ca", "l2c", "l1ca+l2c"],
+                      help="Signal bands to enable",
+                      action=UpdateBands)
+  parser.add_argument('--doppler-type',
+                      default="zero",
+                      choices=["zero", "const", "linear", "sine"],
+                      help="Configure doppler type",
+                      action=UpdateDopplerType)
+  parser.add_argument('--doppler-value',
+                      type=float,
+                      default=-10.,
+                      help="Doppler shift in hertz (initial)",
+                      action=UpdateDopplerType)
+  parser.add_argument('--doppler-speed',
+                      type=float,
+                      default=-10.,
+                      help="Doppler shift change in hertz/second",
+                      action=UpdateDopplerType)
+  parser.add_argument('--doppler-distance',
+                      type=float,
```

Comment:
Some of the arguments have no default value and some do. Might be good to unify the convention if possible.